x-pack/test/encrypted_saved_objects_api_integration/tests/encrypted_saved_objects_decryption.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const esArchiver = getService('esArchiver');
+
+  describe('encrypted saved objects decryption', () => {
+    describe('migrations', () => {
+      before(async () => {
+        await esArchiver.load(
+          'x-pack/test/encrypted_saved_objects_api_integration/fixtures/es_archiver/encrypted_saved_objects_different_key'
+        );
+      });
+
+      after(async () => {
+        await esArchiver.unload(
+          'x-pack/test/encrypted_saved_objects_api_integration/fixtures/es_archiver/encrypted_saved_objects_different_key'
+        );
+      });
+
+      it('migrates alert and actions saved objects even if decryption fails', async () => {
+        const { body: migratedRule } = await supertest
+          .get(`/api/alerting/rule/a0d18560-e985-11eb-b1e3-5b27f0de1e72`)
```

Comment:
__question:__ just for my understanding, do we use `alert` SO type for `rule` entity and `action` SO type for `connector` entity? That confused me a bit.

Reply:
Yes, that is correct and agree that it is confusing! We have [this issue](https://github.com/elastic/kibana/issues/100068) to rename the SO types to reflect the updated alerting terminology